BitwardenShared/UI/Platform/Application/AppProcessor.swift

```diff
@@ -323,3 +324,88 @@ extension AppProcessor: SyncServiceDelegate {
         }
     }
 }
+
+// MARK: - Fido2 credentials
+
+public extension AppProcessor {
+    /// Provides a Fido2 credential for a passkey request.
+    /// - Parameter passkeyRequest: Request to get the credential.
+    @available(iOSApplicationExtension 17.0, *)
+    func provideFido2Credential( // swiftlint:disable:this function_body_length
+        for passkeyRequest: ASPasskeyCredentialRequest
+    ) async throws -> ASPasskeyAssertionCredential {
+        guard let credentialIdentiy = passkeyRequest.credentialIdentity as? ASPasskeyCredentialIdentity else {
+            throw AppProcessorError.invalidOperation
+        }
+
+        let isLocked = try? await services.authRepository.isLocked()
+        let vaultTimeout = try? await services.vaultTimeoutService.sessionTimeoutValue(userId: nil)
+
+        switch (vaultTimeout, isLocked) {
+        case (.never, true):
+            // If the user has enabled Never Lock, but the vault is locked,
+            // unlock the vault before continuing.
+            try await services.authRepository.unlockVaultWithNeverlockKey()
+        case (_, false):
+            break
+        default:
+            break
+        }
```

Comment:
🎨 Unless you're going to add to this, it seems like it would be simpler to use an `if` vs `switch` statement.
```suggestion
        if vaultTimeout == .never, isLocked == true {
            // If the user has enabled Never Lock, but the vault is locked,
            // unlock the vault before continuing.
            try await services.authRepository.unlockVaultWithNeverlockKey()
        }
```

Reply:
In the next PR, there is logic in `default`

Reply:
Ah, makes sense!